api/src/config.rs

```diff
@@ -96,90 +148,3 @@ impl<'de> Deserialize<'de> for ApiKey {
         deserializer.deserialize_struct("ApiKey", FIELDS, ApiKeyVisitor)
     }
 }
-
-#[derive(serde::Deserialize, Clone)]
-pub struct Settings {
-    pub database: PgConnectionConfig,
-    pub application: ApplicationSettings,
-    pub encryption_key: EncryptionKey,
-    pub api_key: String,
-}
-
-#[derive(serde::Deserialize, Clone)]
-pub struct ApplicationSettings {
-    /// host the api listens on
-    pub host: String,
-
-    /// port the api listens on
-    pub port: u16,
-}
-
-impl Display for ApplicationSettings {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        writeln!(f, "    host: {}", self.host)?;
-        writeln!(f, "    port: {}", self.port)
-    }
-}
-
-pub fn get_settings<'a, T: serde::Deserialize<'a>>() -> Result<T, config::ConfigError> {
-    let base_path = std::env::current_dir().expect("Failed to determine the current directory");
-    let configuration_directory = base_path.join("configuration");
-
-    // Detect the running environment.
-    // Default to `dev` if unspecified.
-    let environment: Environment = std::env::var("APP_ENVIRONMENT")
-        .unwrap_or_else(|_| DEV_ENV_NAME.into())
-        .try_into()
-        .expect("Failed to parse APP_ENVIRONMENT.");
-
-    let environment_filename = format!("{}.yaml", environment.as_str());
-    let settings = config::Config::builder()
-        .add_source(config::File::from(
-            configuration_directory.join("base.yaml"),
-        ))
-        .add_source(config::File::from(
-            configuration_directory.join(environment_filename),
-        ))
-        // Add in settings from environment variables (with a prefix of APP and '__' as separator)
-        // E.g. `APP_DESTINATION__BIG_QUERY__PROJECT_ID=my-project-id would set `Settings { destination: BigQuery { project_id }}` to my-project-id
-        .add_source(
-            config::Environment::with_prefix("APP")
-                .prefix_separator("_")
-                .separator("__"),
-        )
-        .build()?;
-
-    settings.try_deserialize::<T>()
-}
-
-pub const DEV_ENV_NAME: &str = "dev";
-pub const PROD_ENV_NAME: &str = "prod";
-
-/// The possible runtime environment for our application.
-pub enum Environment {
```

Comment:
All of the environment logic together with the config loading was moved into the `config` crate.